author: Denis Kristak (Inuits)
correct relative paths in makefiles, add missing includes, prevent unnecessary SW from installing, fix compilation flags
diff -ruN smv-SMV6.7.17_orig/Build/LIBS/intel_linux_64/make_LIBS.make smv-SMV6.7.17/Build/LIBS/intel_linux_64/make_LIBS.make
--- smv-SMV6.7.17_orig/Build/LIBS/intel_linux_64/make_LIBS.make	2021-10-11 15:54:47.661332000 +0200
+++ smv-SMV6.7.17/Build/LIBS/intel_linux_64/make_LIBS.make	2021-10-14 13:59:41.548566000 +0200
@@ -1,18 +1,12 @@
 OPTS="-I $*"
-LIBDIR=$(shell pwd)
-SRCDIR=$(LIBDIR)/../../../Source
+LIBDIR?=$(shell pwd)
+SRCDIR?=$(LIBDIR)/../../../Source
 
-all: libgd.a libglui.a libglut.a libjpeg.a libpng.a libz.a 
-
-# GD
-libgd.a:
-	@echo $(LIBDIR)
-	cd $(SRCDIR)/gd-2.0.15; \
-		./makelib.sh $(OPTS); \
-		cp libgd.a $(LIBDIR)/.
+all: libglui.a libglut.a
 
 # GLUI
 libglui.a:
+	@echo $(LIBDIR);
 	cd $(SRCDIR)/glui_v2_1_beta; \
 		./makelib.sh $(OPTS); \
 		cp libglui.a $(LIBDIR)/.
@@ -24,38 +18,4 @@
 	./makelib.sh $(OPTS); \
 	cp libglut.a $(LIBDIR)/.
 
-# JPEG
-libjpeg.a:
-	cd $(SRCDIR)/jpeg-9b; \
-		./makelib.sh $(OPTS); \
-		cp libjpeg.a $(LIBDIR)/.
-
-# PNG
-libpng.a:
-	cd $(SRCDIR)/png-1.6.21; \
-		./makelib.sh $(OPTS); \
-		cp libpng.a $(LIBDIR)/.
-
-# ZLIB
-libz.a:
-	cd $(SRCDIR)/zlib128; \
-		./makelib.sh $(OPTS); \
-		cp libz.a $(LIBDIR)/.
-
-# Lua # Lua interpreter
-liblua.a:
-	cd $(SRCDIR)/lua-5.3.1; \
-		export TARGET=linux; \
-		./makelib.sh $(OPTS); \
-		cp src/liblua.a $(LIBDIR)/.
-
-# LPEG # Lua parsing libarary to parse SSF files
-# This depends on lua being built first
-lpeg.so: liblua.a
-	cd $(SRCDIR)/lpeg-1.0.0; \
-		pwd; \
-		export TARGET=linux; \
-		./makelib.sh $(OPTS); \
-		cp lpeg.so $(LIBDIR)/.
-
 .PHONY: all
diff -ruN smv-SMV6.7.17_orig/Build/smokeview/Makefile smv-SMV6.7.17/Build/smokeview/Makefile
--- smv-SMV6.7.17_orig/Build/smokeview/Makefile	2021-10-11 15:54:45.856668000 +0200
+++ smv-SMV6.7.17/Build/smokeview/Makefile	2021-10-11 14:19:20.498029000 +0200
@@ -3,8 +3,8 @@
 
 #To use this Makefile cd to a sub-directory and type make_smv.sh or make_smv.bat
 
-SOURCE_DIR = ../../../Source/
-BUILD_DIR = ../../../Build
+SOURCE_DIR = ../../Source/
+BUILD_DIR = ../../Build
 LIB_DIR = $(BUILD_DIR)/LIBS
 
 # Set the C standard we are using. Currently this is just the appropriate flag
@@ -18,27 +18,16 @@
 SMV_PROFILESTRING =
 COMP_VERSION =
 LUA_SCRIPTING =
-ifeq ($(shell echo "check_quotes"),"check_quotes")
-  GIT_HASH   := $(shell ..\..\..\Utilities\Scripts\githash)
-  GIT_DATE   := $(shell ..\..\..\Utilities\Scripts\gitlog)
-  GIT_BRANCH := $(shell ..\..\..\Utilities\Scripts\gitbranch)
-else
-  GLIBVERS   := $(shell echo $(GLIBVERS))
-  GIT_HASH   := $(shell git describe --long --dirty)
-  GIT_DATE   := $(shell git log -1 --format=%cd)
-  GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-endif
-GITINFO=-Dpp_GITHASH=\"$(GIT_HASH)-$(GIT_BRANCH)\" -Dpp_GITDATE=\""$(GIT_DATE)\""
 
 ifeq ($(shell echo "check_quotes"),"check_quotes")
-  INTEL_COMPVERSION := $(shell ..\..\..\Utilities\Scripts\intel_compversion)
-  GNU_COMPVERSION := $(shell ..\..\..\Utilities\Scripts\gnu_compversion)
+	INTEL_COMPVERSION := $(shell ..\..\Utilities\Scripts\intel_compversion)
+	GNU_COMPVERSION := $(shell ..\..\Utilities\Scripts\gnu_compversion)
 else
-  INTEL_COMPVERSION := $(shell ../../../Utilities/Scripts/intel_compversion.sh)
-  GNU_COMPVERSION := $(shell ../../../Utilities/Scripts/gnu_compversion.sh)
+	INTEL_COMPVERSION := $(shell ../../Utilities/Scripts/intel_compversion.sh)
+	GNU_COMPVERSION := $(shell ../../Utilities/Scripts/gnu_compversion.sh)
 endif
-INTEL_COMPINFO=-Dpp_COMPVER=\"$(INTEL_COMPVERSION)\"
-GNU_COMPINFO=-Dpp_COMPVER=\"$(GNU_COMPVERSION)\"
+	INTEL_COMPINFO=-Dpp_COMPVER=\"$(INTEL_COMPVERSION)\"
+	GNU_COMPINFO=-Dpp_COMPVER=\"$(GNU_COMPVERSION)\"
 
 .rc.obj:
 	windres $< -o $@
@@ -84,7 +73,7 @@
 
 objwin = $(obj:.o=.obj)
 
-INC += -I $(SOURCE_DIR)/glui_v2_1_beta -I $(SOURCE_DIR)/gd-2.0.15 -I $(SOURCE_DIR)/shared -I $(SOURCE_DIR)/smokeview -I $(SOURCE_DIR)/glew -I $(SOURCE_DIR)/zlib128
+INC += -I $(SOURCE_DIR)/glui_v2_1_beta -I$(EBROOTLIBGD)/include -I $(SOURCE_DIR)/shared -I $(SOURCE_DIR)/smokeview -I $(EBROOTGLEW)/include/GL -I $(EBROOTZLIB)/include
 
 # windows include directories
 
@@ -175,7 +164,7 @@
 
 intel_win_64_db : INC += $(WININC) -I $(SOURCE_DIR)/shared -I $(SOURCE_DIR)/smokeview
 intel_win_64_db : FFLAGS    = /Od /iface:stdref /fpp -D WIN32 /Zi /nologo /fpscomp:general /debug:full /extend_source:132 /warn:unused /warn:nointerfaces /Qtrapuv /fp:strict /fp:except /traceback /check:all /stand:f18
-intel_win_64_db : CFLAGS    = /Od /W4 /debug:full /Zi -D _DEBUG -D WIN32 -D _CONSOLE -D X64 -D GLEW_STATIC -D PTW32_STATIC_LIB $(SMV_TESTFLAG) $(GITINFO) $(INTEL_COMPINFO)
+intel_win_64_db : CFLAGS    = /Od /W4 /debug:full /Zi -D _DEBUG -D WIN32 -D _CONSOLE -D X64 -D GLEW_STATIC -D PTW32_STATIC_LIB $(SMV_TESTFLAG)  $(INTEL_COMPINFO)
 ifeq ($(GLUT),freeglut)
 intel_win_64_db : CFLAGS   += -D FREEGLUT_STATIC
 endif
@@ -195,7 +184,7 @@
 intel_win_64 : INC += -I $(SOURCE_DIR)/lua-5.3.1/src
 endif
 intel_win_64 : FFLAGS    = -O2 /iface:stdref /fpp -D WIN32 /fpscomp:general /stand:f18
-intel_win_64 : CFLAGS    = -O1 -D WIN32 -D _CONSOLE -D X64 -D GLEW_STATIC -D PTW32_STATIC_LIB $(SMV_TESTFLAG) $(GITINFO) $(INTEL_COMPINFO)
+intel_win_64 : CFLAGS    = -O1 -D WIN32 -D _CONSOLE -D X64 -D GLEW_STATIC -D PTW32_STATIC_LIB $(SMV_TESTFLAG)  $(INTEL_COMPINFO)
 ifeq ($(GLUT),freeglut)
 intel_win_64 : CFLAGS   += -D FREEGLUT_STATIC
 endif
@@ -219,7 +208,7 @@
 
 gnu_win_64 : INC       += -I $(SOURCE_DIR)/glui_gl -I $(SOURCE_DIR)/shared -I $(SOURCE_DIR)/smokeview
 gnu_win_64 : FFLAGS    = -O0 -m64 -x f95-cpp-input -D pp_GCC -ffree-form -frecord-marker=4 $(SMV_PROFILEFLAG)
-gnu_win_64 : CFLAGS    = -O0 -m64 $(D_FREEGLUT) -D GLEW_STATIC -D WIN32 -D pp_GCC -D __MINGW32__ -fno-builtin -Wno-write-strings $(GITINFO) $(GNU_COMPINFO) $(SMV_PROFILEFLAG)
+gnu_win_64 : CFLAGS    = -O0 -m64 $(D_FREEGLUT) -D GLEW_STATIC -D WIN32 -D pp_GCC -D __MINGW32__ -fno-builtin -Wno-write-strings  $(GNU_COMPINFO) $(SMV_PROFILEFLAG)
 gnu_win_64 : LFLAGS    = -m64 $(SMV_LIBS_GNU) -lgfortran -lm -lopengl32 -lglu32 -lgdi32 -lwinmm -lcomdlg32 -lhid -lquadmath $(SMV_PROFILEFLAG)
 gnu_win_64 : CC        = gcc
 gnu_win_64 : CPP       = g++
@@ -241,7 +230,7 @@
 
 gnu_win_64_db : INC       += -I $(SOURCE_DIR)/glui_gl -I $(SOURCE_DIR)/shared -I $(SOURCE_DIR)/smokeview
 gnu_win_64_db : FFLAGS    = -O0 -m64 -x f95-cpp-input -D _DEBUG -D pp_GCC -ffree-form -frecord-marker=4 $(SMV_PROFILEFLAG) 
-gnu_win_64_db : CFLAGS    = -O0 -m64 -g -Wunused-label -Wunused-value -Wunused-variable $(D_FREEGLUT) $(SMV_PROFILEFLAG) -D GLEW_STATIC -D WIN32 -D pp_GCC -D __MINGW32__ -fno-builtin -Wno-write-strings $(GITINFO) $(GNU_COMPINFO)
+gnu_win_64_db : CFLAGS    = -O0 -m64 -g -Wunused-label -Wunused-value -Wunused-variable $(D_FREEGLUT) $(SMV_PROFILEFLAG) -D GLEW_STATIC -D WIN32 -D pp_GCC -D __MINGW32__ -fno-builtin -Wno-write-strings  $(GNU_COMPINFO)
 gnu_win_64_db : LFLAGS    = -m64 $(SMV_LIBS_GNU) -lgfortran -lm -lopengl32 -lglu32 -lgdi32 -lwinmm -lcomdlg32 -lhid -lquadmath $(SMV_PROFILEFLAG)
 gnu_win_64_db : CC        = gcc
 gnu_win_64_db : CPP       = g++
@@ -269,7 +258,7 @@
 
 intel_linux_64 : LIB_DIR_PLAT = $(LIB_DIR)/intel_linux_64
 intel_linux_64 : FFLAGS    = -O0 -traceback -m64 -static-intel -fpp -stand:f18 $(SMV_PROFILEFLAG)
-intel_linux_64 : CFLAGS    = -O0 -traceback -m64 -static-intel -D pp_LINUX $(SMV_TESTFLAG) $(SMV_PROFILEFLAG) $(GITINFO) $(INTEL_COMPINFO)
+intel_linux_64 : CFLAGS    = -O0 -traceback -m64 -static-intel -D pp_LINUX $(SMV_TESTFLAG) $(SMV_PROFILEFLAG)  $(INTEL_COMPINFO)
 intel_linux_64 : LIBLUA    =
 ifeq ($(LUA_SCRIPTING),true)
 intel_linux_64 : CFLAGS    += -D pp_LUA
@@ -283,12 +272,12 @@
 intel_linux_64 : exe       = smokeview_linux_$(SMV_TESTSTRING)64$(SMV_PROFILESTRING)
 
 intel_linux_64 : $(obj) $(if $(LUA_SCRIPTING),smvluacore)
-	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L$(LIB_DIR_PLAT) $(SMV_LIBS_LINUX) $(INTEL_LIBS_LINUX) $(SYSTEM_LIBS_LINUX) $(LIBLUA)
+	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L$(LIB_DIR_PLAT)  -L$(EB_L_PATH)  $(SMV_LIBS_LINUX) $(INTEL_LIBS_LINUX) $(SYSTEM_LIBS_LINUX) $(LIBLUA)
 
 # ------------- intel_linux_64_db ----------------
 
 intel_linux_64_db : FFLAGS    = -O0 -m64 -static-intel -traceback -g -fpe0 -fltconsistency -WB -fpp -stand:f18 $(SMV_PROFILEFLAG)
-intel_linux_64_db : CFLAGS    = -O0 -g -m64 -static-intel $(SMV_TESTFLAG) -D _DEBUG -D pp_LINUX $(SMV_PROFILEFLAG) -traceback -Wall -Wextra -check=stack,uninit -fp-stack-check -fp-trap-all=divzero,invalid,overflow -ftrapuv -Wuninitialized -Wunused-function -Wunused-variable $(GITINFO) $(INTEL_COMPINFO)
+intel_linux_64_db : CFLAGS    = -O0 -g -m64 -static-intel $(SMV_TESTFLAG) -D _DEBUG -D pp_LINUX $(SMV_PROFILEFLAG) -traceback -Wall -Wextra -check=stack,uninit -fp-stack-check -fp-trap-all=divzero,invalid,overflow -ftrapuv -Wuninitialized -Wunused-function -Wunused-variable  $(INTEL_COMPINFO)
 intel_linux_64_db : LFLAGS    = -m64 -static-intel -no-cilk $(INTEL_LIBS_LINUX) -traceback $(SMV_PROFILEFLAG)
 intel_linux_64_db : CC        = icc
 intel_linux_64_db : CPP       = icpc
@@ -296,13 +285,13 @@
 intel_linux_64_db : exe       = smokeview_linux_$(SMV_TESTSTRING)64_db$(SMV_PROFILESTRING)
 
 intel_linux_64_db : $(obj)
-	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L$(LIB_DIR)/intel_linux_64 $(SMV_LIBS_LINUX) $(INTEL_LIBS_LINUX) $(SYSTEM_LIBS_LINUX)
+	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L$(LIB_DIR)/intel_linux_64 $(SMV_LIBS_LINUX) $(INTEL_LIBS_LINUX) $(SYSTEM_LIBS_LINUX) -L$(EB_L_PATH) 
 
 # ------------- gnu_linux_64_db ----------------
 
 gnu_linux_64_db : LIB_DIR_PLAT = $(LIB_DIR)/gnu_linux_64
 gnu_linux_64_db : FFLAGS    = -O0 -m64 -ggdb -Wall -x f95-cpp-input -D pp_GCC -ffree-form -frecord-marker=4 -fcheck=all -fbacktrace $(SMV_PROFILEFLAG)
-gnu_linux_64_db : CFLAGS    = -O0 -m64 -ggdb -Wall -Wno-parentheses -Wno-unknown-pragmas -Wno-comment -Wno-write-strings -D _DEBUG -D pp_LINUX -D pp_GCC $(SMV_TESTFLAG) $(GNU_COMPINFO) $(GITINFO) $(SMV_PROFILEFLAG)
+gnu_linux_64_db : CFLAGS    = -O0 -m64 -ggdb -Wall -Wno-parentheses -Wno-unknown-pragmas -Wno-comment -Wno-write-strings -D _DEBUG -D pp_LINUX -D pp_GCC $(SMV_TESTFLAG) $(GNU_COMPINFO)  $(SMV_PROFILEFLAG)
 ifeq ($(LUA_SCRIPTING),true)
 gnu_linux_64_db : CFLAGS    += -D pp_LUA
 gnu_linux_64_db : SMVLUACORE_FILES += $(LIB_DIR_PLAT)/lpeg.so
@@ -315,14 +304,14 @@
 
 gnu_linux_64_db : $(obj)
 	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L $(LIB_DIR_PLAT) \
-		$(SMV_LIBS_LINUX) -lgfortran $(SYSTEM_LIBS_LINUX) \
+		$(SMV_LIBS_LINUX) -lgfortran $(SYSTEM_LIBS_LINUX) -L$(EB_L_PATH) \
 		$(if $(LUA_SCRIPTING),$(LIB_DIR_PLAT)/liblua.a -ldl)
 
 # ------------- gnu_linux_64 ----------------
 
 gnu_linux_64 : LIB_DIR_PLAT = $(LIB_DIR)/gnu_linux_64
 gnu_linux_64 : FFLAGS    = -O0 -m64 -ggdb -Wall -x f95-cpp-input -D pp_GCC -ffree-form -frecord-marker=4 -fcheck=all -fbacktrace $(SMV_PROFILEFLAG)
-gnu_linux_64 : CFLAGS    = -O0 -m64 -ggdb -Wall -Wno-parentheses -Wno-unknown-pragmas -Wno-comment -Wno-write-strings           -D pp_LINUX -D pp_GCC $(SMV_TESTFLAG) $(GNU_COMPINFO) $(GITINFO) $(SMV_PROFILEFLAG)
+gnu_linux_64 : CFLAGS    = -O0 -m64 -ggdb -Wall -Wno-parentheses -Wno-unknown-pragmas -Wno-comment -Wno-write-strings           -D pp_LINUX -D pp_GCC $(SMV_TESTFLAG) $(GNU_COMPINFO)  $(SMV_PROFILEFLAG)
 ifeq ($(LUA_SCRIPTING),true)
 gnu_linux_64 : CFLAGS    += -D pp_LUA
 gnu_linux_64 : SMVLUACORE_FILES += $(LIB_DIR_PLAT)/lpeg.so
@@ -335,7 +324,7 @@
 
 gnu_linux_64 : $(obj) $(if $(LUA_SCRIPTING),smvluacore)
 	$(CPP) -o $(bin)/$(exe) $(obj) $(LFLAGS) -L $(LIB_DIR_PLAT) \
-		$(SMV_LIBS_LINUX) -lgfortran $(SYSTEM_LIBS_LINUX) \
+		$(SMV_LIBS_LINUX) -lgfortran $(SYSTEM_LIBS_LINUX)  -L$(EB_L_PATH) \
 		$(if $(LUA_SCRIPTING),$(LIB_DIR_PLAT)/liblua.a -ldl)
 
 # ------------- mingw_win_64 ----------------
@@ -385,7 +374,7 @@
 # ------------- intel_osx_64 ----------------
 
 intel_osx_64 : FFLAGS    = -O0 -m64 -traceback -static-intel -fpp -D pp_OSX  -mmacosx-version-min=10.9 -stand:f18
-intel_osx_64 : CFLAGS    = -O0 -m64 -I $(SOURCE_DIR)/glui_gl -traceback -static-intel -D pp_OSX $(SMV_TESTFLAG) $(GITINFO) $(INTEL_COMPINFO) -mmacosx-version-min=10.9
+intel_osx_64 : CFLAGS    = -O0 -m64 -I $(SOURCE_DIR)/glui_gl -traceback -static-intel -D pp_OSX $(SMV_TESTFLAG)  $(INTEL_COMPINFO) -mmacosx-version-min=10.9
 intel_osx_64 : LIBLUA    =
 ifeq ($(LUA_SCRIPTING),true)
 intel_osx_64 : CFLAGS    += -D pp_LUA
@@ -415,7 +404,7 @@
 # ------------- intel_osx_64_db ----------------
 
 intel_osx_64_db : FFLAGS    = -O0 -g -m64 -traceback -static-intel -fpp     -D pp_OSX  -mmacosx-version-min=10.9 -stand:f18
-intel_osx_64_db : CFLAGS    = -O0 -g -m64 -I $(SOURCE_DIR)/glui_gl -traceback -static-intel -D _DEBUG -D pp_OSX $(SMV_TESTFLAG) $(GITINFO) $(INTEL_COMPINFO) -mmacosx-version-min=10.9
+intel_osx_64_db : CFLAGS    = -O0 -g -m64 -I $(SOURCE_DIR)/glui_gl -traceback -static-intel -D _DEBUG -D pp_OSX $(SMV_TESTFLAG)  $(INTEL_COMPINFO) -mmacosx-version-min=10.9
 intel_osx_64_db : LIBLUA    =
 ifeq ($(LUA_SCRIPTING),true)
 intel_osx_64_db : CFLAGS    += -D pp_LUA
@@ -444,7 +433,7 @@
 # ------------- gnu_osx_64_db ----------------
 
 gnu_osx_64_db : FFLAGS    = -O0 -m64 -x f95-cpp-input -D pp_GCC -ffree-form $(SMV_PROFILEFLAG)
-gnu_osx_64_db : CFLAGS    = -O0 -ggdb -m64 -D _DEBUG -D pp_OSX -D pp_GCC $(SMV_TESTFLAG) -Wall -Wno-comment -Wno-deprecated-declarations -Wno-write-strings $(GNU_COMPINFO) $(GITINFO) $(SMV_PROFILEFLAG)
+gnu_osx_64_db : CFLAGS    = -O0 -ggdb -m64 -D _DEBUG -D pp_OSX -D pp_GCC $(SMV_TESTFLAG) -Wall -Wno-comment -Wno-deprecated-declarations -Wno-write-strings $(GNU_COMPINFO)  $(SMV_PROFILEFLAG)
 gnu_osx_64_db : LFLAGS    = -m64 -L$(LIB_DIR)/gnu_osx_64 $(SMV_LIBS_OSX) $(GLIBDIROPT) -lgfortran $(SMV_PROFILEFLAG)
 ifeq ($(GLUT),freeglut)
 gnu_osx_64_db : LFLAGS   += -L /opt/X11/lib -lX11 -lXmu -lGLU -lGL
@@ -467,7 +456,7 @@
 # ------------- gnu_osx_64 ----------------
 
 gnu_osx_64 : FFLAGS    = -O0 -m64 -x f95-cpp-input -D pp_GCC -ffree-form $(SMV_PROFILEFLAG)
-gnu_osx_64 : CFLAGS    = -O0 -m64              -D pp_OSX -D pp_GCC $(SMV_TESTFLAG) -Wall -Wno-comment -Wno-deprecated-declarations -Wno-write-strings $(GNU_COMPINFO) $(GITINFO) $(SMV_PROFILEFLAG)
+gnu_osx_64 : CFLAGS    = -O0 -m64              -D pp_OSX -D pp_GCC $(SMV_TESTFLAG) -Wall -Wno-comment -Wno-deprecated-declarations -Wno-write-strings $(GNU_COMPINFO)  $(SMV_PROFILEFLAG)
 gnu_osx_64 : LFLAGS    = -m64 -L$(LIB_DIR)/gnu_osx_64 $(SMV_LIBS_OSX) $(GLIBDIROPT) -lgfortran $(SMV_PROFILEFLAG)
 ifeq ($(GLUT),freeglut)
 gnu_osx_64 : LFLAGS   += -L /opt/X11/lib -lX11 -lXmu -lGLU -lGL
diff -ruN smv-SMV6.7.17_orig/Source/shared/file_util.c smv-SMV6.7.17/Source/shared/file_util.c
--- smv-SMV6.7.17_orig/Source/shared/file_util.c	2021-10-11 15:54:46.718505000 +0200
+++ smv-SMV6.7.17/Source/shared/file_util.c	2021-10-11 15:19:55.038992000 +0200
@@ -25,6 +25,7 @@
 #include <dirent.h>
 #endif
 #include "MALLOCC.h"
+#include <sys/fcntl.h>
 
 FILE *alt_stdout=NULL;
 
diff -ruN smv-SMV6.7.17_orig/Source/shared/file_util.h smv-SMV6.7.17/Source/shared/file_util.h
--- smv-SMV6.7.17_orig/Source/shared/file_util.h	2021-10-11 15:54:46.718987000 +0200
+++ smv-SMV6.7.17/Source/shared/file_util.h	2021-10-11 15:20:05.595248000 +0200
@@ -11,12 +11,12 @@
 #ifdef pp_GCC
 #include <unistd.h>
 #endif
-#ifdef pp_LINUX
 #include <unistd.h>
-#endif
 #ifndef WIN32
 #include <sys/stat.h>
 #endif
+#include <sys/fcntl.h>
+#include <unistd.h>
 
 // vvvvvvvvvvvvvvvvvvvvvvvv structures vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
 
diff -ruN smv-SMV6.7.17_orig/Source/smokeview/glui_colorbar.cpp smv-SMV6.7.17/Source/smokeview/glui_colorbar.cpp
--- smv-SMV6.7.17_orig/Source/smokeview/glui_colorbar.cpp	2021-10-11 15:54:46.778813000 +0200
+++ smv-SMV6.7.17/Source/smokeview/glui_colorbar.cpp	2021-10-07 15:35:13.135085000 +0200
@@ -168,7 +168,7 @@
       char *clabel;
 
       cbi = colorbarinfo + colorbartype;
-      clabel = EDITTEXT_colorbar_label->get_text();
+      strcpy(clabel, EDITTEXT_colorbar_label->get_text());
       strcpy(cbi->label, clabel);
       LISTBOX_colorbar->delete_item(colorbartype);
       LISTBOX_colorbar->add_item(colorbartype, colorbar_label);
diff -ruN smv-SMV6.7.17_orig/Source/smokeview/main.c smv-SMV6.7.17/Source/smokeview/main.c
--- smv-SMV6.7.17_orig/Source/smokeview/main.c	2021-10-11 15:54:46.838073000 +0200
+++ smv-SMV6.7.17/Source/smokeview/main.c	2021-10-11 15:20:33.131615000 +0200
@@ -5,6 +5,8 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <unistd.h>
+
 #include GLUT_H
 
 #include "string_util.h"
